modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/AbstractCollocationBenchmark.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.benchmark;
+
+import static org.apache.ignite.catalog.definitions.ColumnDefinition.column;
+import static org.apache.ignite.internal.catalog.CatalogService.DEFAULT_STORAGE_PROFILE;
+import static org.apache.ignite.internal.testframework.TestIgnitionManager.PRODUCTION_CLUSTER_CONFIG_STRING;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.ignite.catalog.ColumnType;
+import org.apache.ignite.catalog.definitions.TableDefinition;
+import org.apache.ignite.catalog.definitions.ZoneDefinition;
+import org.apache.ignite.internal.replicator.TablePartitionId;
+import org.apache.ignite.internal.replicator.ZonePartitionId;
+import org.apache.ignite.internal.tx.message.WriteIntentSwitchReplicaRequest;
+import org.apache.ignite.table.KeyValueView;
+import org.apache.ignite.table.Table;
+import org.apache.ignite.table.Tuple;
+
+/**
+ * Base class that allows to measure basic KeyValue operations for tables that share the same distribution zone.
+ * TODO: https://issues.apache.org/jira/browse/IGNITE-22522 remove this benchmark.
+ */
+public class AbstractCollocationBenchmark extends AbstractMultiNodeBenchmark {
+    /** Name of shared zone. */
+    private static final String SHARED_ZONE_NAME = "shared_zone";
+
+    /** System property that allows to enable/disable collocation feature. */
+    private static final String FEATURE_FLAG_NAME = "IGNITE_ZONE_BASED_REPLICATION";
```

Comment:
Can't we directly use the constant from `PartitionReplicaLifecycleManager` here?

Reply:
It seems to me, that we need to move this property to a shared place and use it everywhere. Perhaps, it would be nice to add a "hidden" configuration parameter instead of a system property.